Fix capitalization of df2NumericMatrix.

btoomey opened the pull request:
Note that this will probably go away, and we'll probably coerce strings to numeric variables. However, I wanted to have some sort of working version before the meeting where Todd and Neil will decide how to handle such cases. We may either use model.matrix or the Hill and Smith algorithm.

btoomey committed:
Fix capitalization of the df2NumericMatrix function. Note that this will probably go away, and we'll probably coerce strings to numeric variables. However, I wanted to have some sort of working version before the meeting where Todd and Neil will decide how to handle such cases.

btoomey committed:
Save correct lambda values depending on case (glmnet vs cv.glmnet)

ramnathv commented on R/linearregutils.R:
```diff
@@ -82,15 +82,29 @@ df2NumericMatrix <- function(x){
   )
   the.model <- do.call(glmFun, Filter(Negate(is.null), funParams))
-  the.model$lambda_pred <- config$lambda
+  if (config$cv_glmnet) {
```
Should this not be `config$internal_cv` since you changed it?

btoomey replied:
Yes, you're right! Thanks, good catch!

ramnathv commented on R/linearregutils.R:
```diff
@@ -82,15 +82,29 @@ df2NumericMatrix <- function(x){
+      the.model$lambda_pred <- "lambda.1se"
+    } else {
+      the.model$lambda_pred <- "lambda.min"
+    }
```
In idiomatic R, you would write this as

```r
the.model$lambda_pred <- if (config$lambda_1se) 'lambda.1se' else 'lambda.min'
```

ramnathv replied:
Also, this is exactly the type of logic that would be best served by writing tests. This would ensure that you don't break anything accidentally when making changes at a later stage.

btoomey committed:
Fix name of cv_glmnet.

btoomey committed:
Score elnet and cv.glmnet models generated by the Linear Regression tool.

btoomey committed:
Add config$ to lambda_no_cv, since it's an element of the list config.

btoomey committed:
Use predict::glmnet instead of library(glmnet) followed by predict

btoomey committed:
Create plot outputs.

btoomey committed:
Create skeleton of Report function for glmnet.

ramnathv commented on R/linearregutils.R:
```diff
@@ -141,3 +159,51 @@ createPlotOutputsLinearOSR <- function(the.model){
+  }
+  plot(the.model, xvar = "dev")
+}
+
```
This is a perfect example of where you want to use the `DRY` principle

```r
createPlotOutputsGLMNET <- function(the.model, xvar, ...){
  plot_obj <- if (inherits(the.model, "cv.glmnet")) the.model$glmnet.fit else the.model
  function(){plot(plot_obj, xvar = xvar, ...)}
}
```

ramnathv commented on R/runModels.R:
```diff
@@ -95,8 +95,19 @@ getResultsLinearRegression <- function(inputs, config){
+    } else {
+      results <- list(model = the.model, report = NULL, first_plot = plot.one,
+                      second_plot = plot.two, third_plot = plot_three)
+    }
```
```r
results <- append(
  plyr::llply(c('norm', 'lambda', 'dev'), createPlotOutputsGLMNET),
  list(the.model = the.model)
)
if (config$internal_cv) {
  results <- append(results, list(plot.internalcv = plot(the.model)))
}
```

btoomey committed:
Incorporate Ramnath's suggestions for shortening plotting code, create a data.frame of coefficients, and write everything out.

btoomey commented:
@ramnathv : I incorporated your suggestion and added some other things. Assuming the build check passes, can you merge it into the build? Everything is pretty much complete from a first pass point of view. (Though re-factoring will probably ensue.)

ramnathv commented:
@btoomey This has been merged. There are no tests in place for the regularization stuff (either R unit tests or functional tests). I would strongly recommend that you add some so that every time we merge something into the build, we can track there are no regressions of any kind.
